Library/Homebrew/test/migrator_spec.rb

```diff
@@ -0,0 +1,269 @@
+require "migrator"
+require "test/support/fixtures/testball"
+require "tab"
+require "keg"
+
+class Migrator
+  describe self do
+    let(:new_formula) { Testball.new("newname") }
+    let(:old_formula) { Testball.new("oldname") }
+
+    before(:each) do
+      allow(new_formula).to receive(:oldname).and_return("oldname")
+    end
+
+    it "raises an error if there is no old name" do
+      expect {
+        described_class.new(old_formula)
+      }.to raise_error(MigratorNoOldnameError)
+    end
+
+    it "raises an error if there is no old path" do
+      expect {
+        described_class.new(new_formula)
+      }.to raise_error(MigratorNoOldpathError)
+    end
+
+    it "raises an error if the Taps differ" do
+      keg = HOMEBREW_CELLAR/"oldname/0.1"
+      keg.mkpath
+      tab = Tab.empty
+      tab.tabfile = HOMEBREW_CELLAR/"oldname/0.1/INSTALL_RECEIPT.json"
+      tab.source["tap"] = "homebrew/core"
+      tab.write
+
+      expect {
+        described_class.new(new_formula)
+      }.to raise_error(MigratorDifferentTapsError)
+    end
+  end
+
+  describe self do
+    subject { described_class.new(new_formula) }
+
+    let(:new_formula) { Testball.new("newname") }
+    let(:old_formula) { Testball.new("oldname") }
+
+    let(:new_keg_record) { HOMEBREW_CELLAR/"newname/0.1" }
+    let(:old_keg_record) { HOMEBREW_CELLAR/"oldname/0.1" }
+
+    let(:old_tab) { Tab.empty }
+
+    let(:keg) { Keg.new(old_keg_record) }
+    let(:old_pin) { HOMEBREW_PINNED_KEGS/"oldname" }
+
+    before(:each) do
+      allow(new_formula).to receive(:oldname).and_return("oldname")
+
+      (old_keg_record/"bin").mkpath
+
+      %w[inside bindir].each do |file|
+        FileUtils.touch old_keg_record/"bin/#{file}"
+      end
+
+      old_tab.tabfile = HOMEBREW_CELLAR/"oldname/0.1/INSTALL_RECEIPT.json"
+      old_tab.source["path"] = "/oldname"
+      old_tab.write
+
+      keg.link
+      keg.optlink
+
+      old_pin.make_relative_symlink old_keg_record
+
+      subject # needs to be evaluated eagerly
+
+      (HOMEBREW_PREFIX/"bin").mkpath
+    end
+
+    after(:each) do
+      if !old_keg_record.parent.symlink? && old_keg_record.directory?
+        keg.unlink
+      end
+
+      if new_keg_record.directory?
+        new_keg = Keg.new(new_keg_record)
+        new_keg.unlink
+      end
+    end
+
+    specify "test_move_cellar" do
```

Comment:
I think these tests need more descriptive names. It's very difficult for me to tell what's going on.